### DISCUSSION TITLE
Transform without labels in segmentation

### DISCUSSION THREAD

**Discussion by talhaanwarch** [Project-MONAI/MONAI#2600] [Q&A]
When we want to predict on test data without labels (mask), how to transform in that case. 

I think it should be something like this
```
train_transforms = T.Compose(
            [
        T.LoadImaged(keys=["image"]),
        T.AddChanneld(keys=["image"]),
        T.Spacingd(keys=["image"],pixdim=(1.5, 1.5, 2.0),mode=("bilinear", "nearest")),
        T.ScaleIntensityd(keys=["image"]),
        T.CropForegroundd(keys=["image"], source_key="image"),
        T.RandCropByPosNegLabeld(
            keys=["image"],
            label_key="label",
            spatial_size=(64, 64, 64),
            pos=1,
            neg=1,
            num_samples=8,
            image_key="image",
            image_threshold=0,
        ),
        T.ToTensord(keys=["image"]),
            ]
        )
```
But how `CropForegroundd` and `RandCropByPosNegLabeld` can know that its foreground and its pos neg label without having a label? 
One way is to remove these two transforms.

I removed `RandCropByPosNegLabeld` as in spleen tutorial before removing this transform, my result looks like (a bit acceptable).
(The following show transform with both image and label)
The first image is ground truth, the second inverted ground truth, third is predicted in each row. 
![image](https://user-images.githubusercontent.com/37379131/125625497-5e7bfdbc-26a4-4160-b66e-1c918fa37969.png)
After removing `RandCropByPosNegLabeld` transform techniques results become horrible like this
![image](https://user-images.githubusercontent.com/37379131/125625770-09e1f756-8f9a-4a3d-b96e-46f9e1498609.png)

**Comment by rijobro** [comment]
Cropping the foreground is something you can only do if you have the ground truth. So at training time, you can use them, but at validation/testing time, you'll probably want to use a sliding window. In this way, you'll still process images that are the same size as they were at training time, but now you can do the whole image. The spleen segmentation tutorial has examples of this.

**Reply by talhaanwarch** [reply]
I followed the spleen tutorial, and used `sliding_window_inference` but you can see that results become very bad when I am not using `RandCropByPosNegLabeld` during inference

**Reply by talhaanwarch** [reply]
do `CropForegroundd` also need ground truth?

**Reply by Nic-Ma** [reply]
Hi @talhaanwarch ,

`CropForegroundd` transform uses the `select_fn` arg to determine foreground:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/croppad/dictionary.py#L803

Thanks.

**Comment by talhaanwarch** [comment]
issue of bad prediction after `RandCropByPosNegLabeld ` is because of using small roi of (32,32,32). Making it (160,160,160) fixed the issue
